website/content/en/docs/best-practices/observability-best-practices.md

```diff
@@ -0,0 +1,255 @@
+---
+title: "Operator Observability Best Practices"
+linkTitle: "Observability Best practices"
+weight: 6
+description: This guide describes the best practices concepts for adding Observability to operators.
+---
+
+## Operator Observability Best Practices
+
+In this document we will outline what operators require in order to meet the "Deep Insights" capability level and provide best practices and examples for creating metrics, recording rules and alerts. It is based on the general guidelines in [Operator Capability Levels](https://sdk.operatorframework.io/docs/overview/operator-capabilities/).
+
+**Note:** For technical documentation of how to add metrics to your operator, please read the [Metrics](https://book.kubebuilder.io/reference/metrics.html) section of the Kubebuilder documentation.
+
+### Deep Insights capability level requirements
+
+1. **Health and Performance metrics** for all of the operator components - Implemented based on the guidelines below.
+2. **Metrics Documentation** - All metrics should have documentation.
+3. **Alerts** for when things are not working as expected for each of the operator's components - Implemented based on the guidelines below.
+4. **Alerts Runbooks** - Each alert MUST include a runbook_url annotation and a runbook.
+5. **Alerts and Metrics Tests** - E2E Testing for metrics and alerts and unit tests for alerts.
+6. **Events** - Custom Resources MUST emit custom events for the operations taking place.
```

Comment:
We should NOT emit events for all Operations. If you raise too many events it brings bad UX experiences for those that are consuming the solutions on the cluster and they will not have attention to what they actually must be notified. 

Events should be raised in certain circumstances only and following this guideline acctually is not a good practice. See that the Kubernetes APIs convention describes when use them here: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#events

> Events are complementary to status information, since they can provide some historical information about status and occurrences in addition to current or previous status. **Generate events for situations users or administrators should be alerted about.**

See that is recommended properly follow the Kubernetes API convention and use conditional. status. Therefore, the cluster admin will be able to check the CR status and know the status of the desired state on the cluster. See: 
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties

@sradco ^